types/director/index.d.ts

```diff
@@ -0,0 +1,549 @@
+        readonly mkcol: typeof _GeneratedHelperMethod;
+        readonly copy: typeof _GeneratedHelperMethod;
+        readonly move: typeof _GeneratedHelperMethod;
+        readonly lock: typeof _GeneratedHelperMethod;
+        readonly unlock: typeof _GeneratedHelperMethod;
+        readonly ["version-control"]: typeof _GeneratedHelperMethod;
+        readonly report: typeof _GeneratedHelperMethod;
+        readonly checkout: typeof _GeneratedHelperMethod;
+        readonly checkin: typeof _GeneratedHelperMethod;
+        readonly uncheckout: typeof _GeneratedHelperMethod;
+        readonly mkworkspace: typeof _GeneratedHelperMethod;
+        readonly update: typeof _GeneratedHelperMethod;
+        readonly label: typeof _GeneratedHelperMethod;
+        readonly merge: typeof _GeneratedHelperMethod;
+        readonly ["baseline-control"]: typeof _GeneratedHelperMethod;
+        readonly mkactivity: typeof _GeneratedHelperMethod;
+        readonly orderpatch: typeof _GeneratedHelperMethod;
+        readonly acl: typeof _GeneratedHelperMethod;
+        readonly search: typeof _GeneratedHelperMethod;
+        readonly patch: typeof _GeneratedHelperMethod;
+        readonly before: typeof _GeneratedHelperMethod;
+        readonly after: typeof _GeneratedHelperMethod;
+    }
+
+    /**
+     * Adds a new `route` to this instance for a specific method and an empty
+     * `path`.
+     * @param route Handler for the method and path
+     */
+    function _GeneratedHelperMethod(this: Router, route: RouteEntry<HttpRouterContext>): void;
+
+    /**
+     * Adds a new `route` to this instance for a specific method and `path`.
+     * @param path Path to set this route on
+     * @param route Handler for the method and path
+     */
+    function _GeneratedHelperMethod(this: Router, path: string | RegExp, route: RouteEntry<HttpRouterContext>): void;
+
+    /**
+     * Adds a new `route` to this instance for a specific method and `path`.
+     * @param path Path to set this route on
+     * @param options Additional options for this route
+     * @param route Handler for the method and path
+     */
+    function _GeneratedHelperMethod(
+        this: Router,
+        path: string | RegExp,
+        options: HttpRouteHandlerOptions | undefined | null,
+        route: RouteEntry<HttpRouterContext>,
+    ): void;
+
+    /**
+     * Names of helper methods dynamically exposed by the HTTP Router.
+     */
+    type HttpRouterGeneratedMethodNames = typeof methods[number] | "before" | "after";
+
+    /**
+     * Helper interface that checks if any dynamically generated helper methods
+     * were accidentally omitted from the definition of `director.http.Router`.
+     */
+    type HttpRouterGeneratedMethodsCheck = {
+        [key in HttpRouterGeneratedMethodNames]: typeof _GeneratedHelperMethod;
+    };
+
+    // HTTP methods
+    // Defined in /lib/director/http/methods.js
+
+    const methods: readonly [
```

Comment:
If you look at [`methods.js`](https://github.com/flatiron/director/blob/master/lib/director/http/methods.js), director actually exports this array. Admittedly there is no practical use for it, since it is only used to bootstrap the helper methods during class initialization.